yt_dlp/extractor/patreon.py

```diff
@@ -227,7 +371,36 @@ def _entries(self, campaign_id, user_id):
 
     def _real_extract(self, url):
 
-        user_id = self._match_id(url)
-        webpage = self._download_webpage(url, user_id, headers={'Cookie': '.'})
-        campaign_id = self._search_regex(r'https://www.patreon.com/api/campaigns/(\d+)/?', webpage, 'Campaign ID')
-        return self.playlist_result(self._entries(campaign_id, user_id), playlist_title=user_id)
+        campaign_id, vanity = self._match_valid_url(url).groups()
+        if campaign_id is None:
+            webpage = self._download_webpage(url, vanity, headers={'User-Agent': self.USER_AGENT})
+            campaign_id = self._search_regex(r'https://www.patreon.com/api/campaigns/(\d+)/?', webpage, 'Campaign ID')
+
+        campaign_response = self._call_api(
+            f'campaigns/{campaign_id}', campaign_id,
+            note='Downloading campaign info', fatal=False,
+            query={'fields[user]': 'image_url,full_name,url'}) or {}
+
+        campaign_info = campaign_response.get('data') or {}
+        channel_name = traverse_obj(campaign_info, ('attributes', 'name'))
+        user_info = traverse_obj(
+            campaign_response, ('included', lambda _, v: v['type'] == 'user'),
+            default={}, expected_type=dict, get_all=False)
+
+        return {
+            '_type': 'playlist',
+            'id': campaign_id,
+            'title': channel_name,
+            'entries': self._entries(campaign_id),
+            'description': clean_html(traverse_obj(campaign_info, ('attributes', 'summary'))),
+            'channel_url': traverse_obj(campaign_info, ('attributes', 'url')),
+            'channel_follower_count': traverse_obj(campaign_info, ('attributes', 'patron_count')),
+            'channel_id': campaign_id,
+            'channel': channel_name,
+            'uploader_url': traverse_obj(user_info, ('attributes', 'url')),
```

Comment:
The urls are same but id/name is different b/w uploader and channel?

Reply:
They are different things -  it is a bit confusing since patreon kinda mixes them together. For reference, my mapping is that channel is a campaign and uploader is a user.

The user url is coming directly from the API from attributes of a user type. For general users, this url is usually of the form `https://www.patreon.com/user?u=<user id>`. But it seems when a user has a campaign, such url seems to be of that of the campaign url.

Interesting observation: you can still access a channel that has a campaign through the user url, and it will sometimes redirect to the campaign url. Other times it'll just load the campaign page. 

So tl;dr this is expected